### DISCUSSION TITLE
Questions de débutant

### DISCUSSION THREAD

**Discussion by coteph** [witsa/synapps#570] [Q&A]
Bonjour,
J'ai 2 problèmes avec des choix et affichage avec du texte
1/ Par exemple la valeur issue d'une variable Modbus prend 4 valeurs pour des vitesses de ventilation.
Le problème est que 0 n'est pas arrêt mais automatique et que 100% n'est pas V3 mais Boost etc
Donc, je désire afficher les différents textes en fonction de la valeur de la variable. La valeur analogique est bien rapatriée et affichée sur mon composite sous Synapps mais cela ne correspond pas a la réalité

2eme problème mais dans l'autre sens, je désire faire un choix de texte ( Liste de choix par exemple) et ensuite écrire dans une ressource une valeur analogique. même si je mets une valeur différentes sous chaque choix, je n'arrive pas a la récupérer sur le Redy 
idem, la table Modbus m'impose par exemple 0 : arrêt, 2 : auto, 3 : PV, 4 : GV, 5 : boost
Merci de votre aide
Philippe

**Comment by WITNicolas** [comment]
Bonjour,

Pour la lecture, il suffit de lier l'Acteur Texte à la variable Modbus.
Une fois que c'est réalisé, sur ce même lien, il faut rajouter un evenement OnReadTransform (Clic, ajouter OnReadTransform)
Clic ensuite sur le lien Jaune qui apparait.

Puis dans la nouvelle fenêtre qui s'ouvre, coller ceci :
  var level = context.value;
  switch(level) {
      case 4:
          return "Texte pour 4";
          break ;
      case 3:
          return "Texte pour 3";
          break;
      case 2:
          return "Texte pour 2";
          break;
      case 1:
          return "Texte pour 1";
          break;
      case 0:
          return "Texte pour 0";
          break;
      default:
          return "?";
          break;
  }

Pour l'écriture à partir d'une boite déroulante, il faut utiliser l'acteur Liste Déroulante.
Dans les options de cet acteur on peut saisir les libellés ainsi que la valeur associée que va prendre l'option choisie.
Il suffit par la suite de liaisonner (Spécifique / Valeur) le Inlink de votre modbus en précisant Ecriture dans votre liaison.

Nicolas.
